azext_edge/tests/edge/init/test_init_int.py

```diff
@@ -0,0 +1,167 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from enum import Enum
+from typing import Dict, List
+import pytest
+from os import mkdir
+
+from azext_edge.edge.providers.orchestration.common import KEYVAULT_ARC_EXTENSION_VERSION
+from ...helpers import run
+from ...generators import generate_names
+
+class ResourceKeys(Enum):
+    custom_location = "Microsoft.ExtendedLocation/customLocations"
+    mq = "Microsoft.IoTOperationsMQ/mq"
+    orchestrator = "Microsoft.IoTOperationsOrchestrator/targets"
+    connected_cluster = "Microsoft.Kubernetes/connectedClusters"
+
+
+@pytest.fixture(scope="function")
+def cluster_cleanup(cluster_setup, settings, tracked_files):
+    from ...settings import EnvironmentVariables
+    settings.add_to_config(EnvironmentVariables.rg.value)
+    settings.add_to_config(EnvironmentVariables.kv.value)
+    settings.add_to_config(EnvironmentVariables.cluster.value)
+    settings.add_to_config(EnvironmentVariables.sp_app_id.value)
+    settings.add_to_config(EnvironmentVariables.sp_object_id.value)
+
+    cert_dir = "certificates"
+    try:
+        mkdir(cert_dir)
+        tracked_files.append(cert_dir)
+    except FileExistsError:
+        pass
+    
+    yield {
+        "clusterName": settings.env.azext_edge_cluster,
+        "resourceGroup": settings.env.azext_edge_rg,
+        "keyVault": settings.env.azext_edge_kv,
```

Comment:
Questions/Ideas for debate:
1. should we try to create a keyvault if not provided? This will significantly increase the time to run the test.
2. if we answer no to 1 -> I plan on using this variable as a switch for whether the test will run (so if no key vault is provided, the test won't run)